app/services/google/parse_service.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Google
+  class ParseService
+    Result = Struct.new(:search_entries, :links_count, :result_page_html)
+    SearchEntry = Struct.new(:kind, :urls, :position)
+
+    def initialize(html)
+      @html = html
+      @doc = Nokogiri::HTML(html)
+    end
+
+    def call
+      Result.new(
+        search_entries: [*top_ads, *bottom_ads, *non_ads_normal, *non_ads_video],
+        links_count: @doc.css('a[href]').count,
+        result_page_html: @html
+      )
+    end
+
+    private
+
+    def top_ads
+      @doc.css('#taw div[data-text-ad]').map do |ad_div|
```

Comment:
Avoid hard-coding HTML selectors in method calls. For code sicoverabaility, it would be best to define these as constants.

Reply:
Fixed in https://github.com/Goose97/google-scraper-ruby/pull/20/commits